public/js/updateStudentProfile.js

```diff
@@ -0,0 +1,69 @@
+const edaidEditButton = document.querySelector('#edaidEditButton')
+const upfrontEditButton = document.querySelector('#upfrontEditButton')
+const laptopEditButton = document.querySelector('#laptopEditButton')
+const githubUsernameEditButton = document.querySelector('#githubUsernameEditButton')
+
+function handleEditClick(event) {
+    const buttonName = event.target.getAttribute('id')
+    const divName = buttonName.replace('EditButton', 'Container')
+    const containerDiv = document.querySelector('.' + divName)
+    console.log(containerDiv)
+    const fieldName = divName.replace('Container', '')
+    const description = buttonName.replace('EditButton', 'Description')
+    const descriptionTag = document.querySelector('#' + description)
+    if(divName == 'laptopContainer') {
+    containerDiv.innerHTML = 
+        '<form class="form studentProfileEditableField">' +
+        '<label>Laptop required: </label>' +
+        '<div>' +
+        '<input type="radio" value="No" id="noLaptop" name="' + fieldName + '">' +
+        '<label for="noLaptop">No</label>' +
+        '<input type="radio" value="Yes" id="yesLaptop" name="' + fieldName + '">' +
+        '<label for="yesLaptop">Yes</label>' +
+        '</div>' +
+        '<input class="saveButton btn btn-sm btn-primary" type="submit" value="Save">' +
+        '</form>'
+    } else {
+        containerDiv.innerHTML = 
+        '<form class="form studentProfileEditableField">' +
+        '<label for="' + fieldName + 'TextBox">' + descriptionTag.textContent + '</label>' +
+        '<input type="text" id="' + fieldName + 'TextBox" name="' + fieldName + '">' + 
+        '<input class="saveButton  btn btn-primary btn-sm" type="submit" value="Save">' +
+        '</form>'
+    }
+
+    const saveButton = document.querySelector('.saveButton')
+    const form = document.querySelector('.form')
```

Comment:
so you're using a class called just "form"? i think you need a more descriptive name